### PR TITLE
rusk-wallet: Fix moonlight withdraw function by not asking for amount

### LINKED ISSUE #2523: rusk-wallet: Moonlight stake reward withdrawal does not work
**Describe the bug**
The behavior between Moonlight and Phoenix reward withdrawal is different. When doing a Phoenix withdrawal of staking rewards, a user only has to fill in the gas limit and gas price.

In Moonlight, the user also needs to provide an amount, which can only be between 1 LUX and their maximum Moonlight balance.

**To Reproduce**
1. Take a wallet with staking rewards
2. Go to Moonlight Transactions -> Moonlight Withdraw
3. Try to run the transaction
4. The transaction should fail/do nothing

**Expected behaviour**
Like the Phoenix withdrawal, the user should get all their accumulated staking rewards.

### PR DESCRIPTION
**Pull request by Daksh14**
Closes #2523

### CODE REVIEW

File: rusk-wallet/src/bin/command.rs [hunk 1]
```diff
@@ -191,11 +191,11 @@ pub(crate) enum Command {
...
         /// Function name to call
         #[clap(short, long)]
-        fn_name: String,
+        name: String,
 
         /// Function arguments for this call
-        #[clap(short, long)]
-        fn_args: Vec<u8>,
+        #[clap(short = 'f', long)]
+        param: Vec<u8>,
```
Comment (Neotamandua) [inline rusk-wallet/src/bin/command.rs:198]:
```suggestion
        args: Vec<u8>,
```
I would name this arg or args. The doc comment above still says arguments, and there is a semantic difference between parameter & argument.

File: rusk-wallet/src/bin/command.rs [hunk 2]
```diff
@@ -191,11 +191,11 @@ pub(crate) enum Command {
 
         /// Function name to call
         #[clap(short, long)]
-        fn_name: String,
+        name: String,
 
         /// Function arguments for this call
-        #[clap(short, long)]
-        fn_args: Vec<u8>,
+        #[clap(short = 'f', long)]
```
Comment (moCello) [inline rusk-wallet/src/bin/command.rs:197]: I'm not familiar with clap, what does this do?

File: rusk-wallet/src/bin/command.rs [hunk 3]
```diff
@@ -351,11 +347,11 @@ pub(crate) enum Command {
...
+        param: Vec<u8>,
```
Comment (Neotamandua) [inline rusk-wallet/src/bin/command.rs:354]: same here

**Review by moCello — CHANGES_REQUESTED** [review]
Please don't change the variables for the contract-calls. They match the struct's fields names and if you change them in one place you should change them everywhere. https://github.com/dusk-network/rusk/blob/master/execution-core/src/transfer/data.rs#L68

File: rusk-wallet/src/bin/command.rs (hunk 2 above)
Comment (Daksh14) [reply rusk-wallet/src/bin/command.rs:197]: There was a flag name conflict that's why I changed the names they should mean the same thing downstream

**Review by Neotamandua — APPROVED** [review]
LGTM

**Review by moCello — APPROVED** [review]

**Review by HDauven — APPROVED** [review]
Awesome, thank you. This does indeed seem to fix the problem for me.

With relation to users being allowed to withdraw a specified amount. I think there's some merit to keeping/introducing this functionality for both Phoenix and Moonlight withdrawals. 

I've added a follow-up issue here that we can tackle later: https://github.com/dusk-network/rusk/issues/2538

**Pull request merged**